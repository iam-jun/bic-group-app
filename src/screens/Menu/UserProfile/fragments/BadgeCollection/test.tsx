import React from 'react';
import {
  View, Dimensions, FlatList, StyleSheet,
} from 'react-native';
import { ExtendedTheme, useTheme } from '@react-navigation/native';
import Avatar from '~/baseComponents/Avatar';
import Icon from '~/baseComponents/Icon';
import spacing, { borderRadius } from '~/theme/spacing';

const Grid = ({ data }: {data: any[]}) => {
  const deviceWidth = Dimensions.get('window').width;

  const itemWidth = 48; // Constant item width
  const numColumns = Math.floor((deviceWidth - 10 * 2) / (itemWidth + 6 * 2));
  const rowWidth = numColumns * (itemWidth + 6 * 2);
  const paddingLeft = (deviceWidth - (10 * 2) - rowWidth) / 2;

  const theme: ExtendedTheme = useTheme();
  const { colors } = theme;
  const styles = themeStyles(theme);

  const isSelected = false;

  const renderGridItem = ({ item, index }: any) =>
    // Render each grid item here
    (
      <View
        key={index}
        style={{
          margin: 6,
          borderWidth: isSelected ? 2 : 1,
          borderColor: isSelected ? colors.purple50 : colors.neutral5,
          borderRadius: borderRadius.pill,
        }}
      >
        <Avatar.Medium
          isRounded
          source={{ uri: item?.iconUrl }}
        />
        {isSelected
          ? (
            <View style={styles.iconChose}>
              <Icon size={10} icon="Check" tintColor={colors.white} />
            </View>
          )
          : null}
      </View>
    );

  return (
    <View style={{
      flex: 1,
      justifyContent: 'center',
      alignItems: 'center',
      paddingHorizontal: 10,
      paddingBottom: 10,
    }}
    >

      <View style={{
        // paddingLeft,
      }}
      >
        <FlatList
          data={data}
          renderItem={renderGridItem}
          keyExtractor={(item, index) => index.toString()}
          numColumns={numColumns}
        />
      </View>
    </View>
  );
};

const themeStyles = (theme: ExtendedTheme) => {
  const { colors } = theme;

  return StyleSheet.create({
    container: {
      flex: 1,
      paddingTop: spacing.padding.large,
    },
    iconChose: {
      backgroundColor: colors.purple50,
      borderRadius: borderRadius.pill,
      height: 20,
      width: 20,
      alignItems: 'center',
      justifyContent: 'center',
      position: 'absolute',
      zIndex: 1,
      right: 0,
    },
  });
};

export default Grid;
