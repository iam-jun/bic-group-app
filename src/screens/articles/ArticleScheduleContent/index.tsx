import { ExtendedTheme, useTheme } from '@react-navigation/native';
import { t } from 'i18next';
import React, { FC, useEffect, useState } from 'react';
import {
  ActivityIndicator, FlatList, RefreshControl, StyleSheet, View,
} from 'react-native';
import { EdgeInsets, useSafeAreaInsets } from 'react-native-safe-area-context';
import Text from '~/baseComponents/Text';
import Header from '~/beinComponents/Header';
import Image from '~/components/Image';
import ScreenWrapper from '~/beinComponents/ScreenWrapper';
import ViewSpacing from '~/beinComponents/ViewSpacing';
import { ArticleScheduleItem } from '~/components/articles';
import { IPayloadGetArticleScheduleContent } from '~/interfaces/IArticle';
import { IRouteParams } from '~/interfaces/IRouter';
import images from '~/resources/images';
import useNetworkStore from '~/store/network';
import networkSelectors from '~/store/network/selectors';
import { spacing } from '~/theme';
import useArticleScheduleContentStore, { IArticleScheduleContentState } from './store';

const ArticleScheduleContent: FC<IRouteParams> = (props) => {
  const groupId = props?.route?.params?.groupId;
  const theme: ExtendedTheme = useTheme();
  const { colors } = theme;
  const insets = useSafeAreaInsets();
  const styles = createStyle(theme, insets);
  const isInternetReachable = useNetworkStore(networkSelectors.getIsInternetReachable);

  const [lossInternet, setLossInternet] = useState(false);

  const {
    articles, hasNextPage, refreshing, actions,
  } = useArticleScheduleContentStore(
    (state: IArticleScheduleContentState) => state,
  );

  useEffect(() => {
    getData(true);
    return () => {
      actions.clearArticleScheduleContent();
    };
  }, []);

  useEffect(() => {
    if (isInternetReachable) {
      if (lossInternet) {
        setLossInternet(false);
        getData(false);
      }
    } else {
      setLossInternet(true);
    }
  }, [isInternetReachable]);

  const onLoadMore = () => {
    getData(false);
  };

  const onRefresh = () => {
    getData(true);
  };

  const getData = (isRefreshing?: boolean) => {
    const payload: IPayloadGetArticleScheduleContent = {
      isRefresh: isRefreshing,
      groupId,
    };
    actions.getArticleScheduleContent(payload);
  };

  const renderItem = ({ item }: any) => <ArticleScheduleItem data={item} isAdmin />;

  const renderFooter = () => (
    <View>
      {hasNextPage && !refreshing && (
        <View testID="draft_article.load_more_view" style={styles.listFooter}>
          <ActivityIndicator color={colors.gray20} />
        </View>
      )}
      {!refreshing && !hasNextPage && <ViewSpacing height={spacing.margin.large} />}
    </View>
  );

  const renderEmpty = () => {
    if (articles.length === 0 && refreshing) {
      return null;
    }

    return (
      <View testID="draft_article.empty_view" style={styles.emptyContainer}>
        <Image style={styles.imgEmpty} source={images.img_empty_box} />
        <Text.BodyS useI18n color={colors.neutral40}>
          your_content:text_empty
        </Text.BodyS>
      </View>
    );
  };

  return (
    <ScreenWrapper testID="article_schedule_content" isFullView style={styles.container}>
      <Header title={t('settings:title_schedule_content')} />
      <FlatList
        testID="article_schedule_content.list"
        style={styles.listContainer}
        data={articles}
        renderItem={renderItem}
        ListHeaderComponent={() => <ViewSpacing height={spacing.margin.large} />}
        ListFooterComponent={renderFooter}
        ListEmptyComponent={renderEmpty}
        ItemSeparatorComponent={() => <ViewSpacing height={spacing.margin.large} />}
        onEndReached={onLoadMore}
        onEndReachedThreshold={0.2}
        refreshControl={<RefreshControl refreshing={refreshing} onRefresh={onRefresh} tintColor={colors.gray40} />}
        keyExtractor={(item, index) => `list-article-schedule-content-${item.id}-${index}`}
      />
    </ScreenWrapper>
  );
};

const createStyle = (theme: ExtendedTheme, insets: EdgeInsets) => {
  const { colors } = theme;
  return StyleSheet.create({
    container: {
      flex: 1,
      backgroundColor: colors.gray5,
      paddingBottom: insets.bottom,
    },
    listContainer: {
      flex: 1,
    },
    listFooter: {
      height: 150,
      justifyContent: 'center',
      alignItems: 'center',
    },
    emptyContainer: {
      paddingVertical: spacing.padding.extraLarge * 2,
      justifyContent: 'center',
      alignItems: 'center',
    },
    imgEmpty: {
      width: 100,
      height: 84,
      marginBottom: spacing.margin.large,
    },
  });
};

export default ArticleScheduleContent;
