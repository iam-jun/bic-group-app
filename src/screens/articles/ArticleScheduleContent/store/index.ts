import { createStore, resetStore } from '~/store/utils';
import getArticleScheduleContent from './actions/getArticleScheduleContent';
import { IPost } from '~/interfaces/IPost';
import IBaseState from '~/store/interfaces/IBaseState';
import IFetchingState from '~/store/interfaces/IFetchingState';
import { IPayloadGetArticleScheduleContent } from '~/interfaces/IArticle';

export interface IArticleScheduleContentState extends IBaseState, IFetchingState {
  articles: IPost[];

  actions?: {
    getArticleScheduleContent: (payload: IPayloadGetArticleScheduleContent) => void;
    clearArticleScheduleContent: () => void;
  };
}

const initialState: IArticleScheduleContentState = {
  articles: [],
  hasNextPage: true,
  loading: false,
  refreshing: false,
};

const articleScheduleContentStore = (set, get) => ({
  ...initialState,

  actions: {
    getArticleScheduleContent: getArticleScheduleContent(set, get),
    clearArticleScheduleContent: () => {
      set((state: IArticleScheduleContentState) => {
        state.articles = initialState.articles;
        state.hasNextPage = initialState.hasNextPage;
        state.loading = initialState.loading;
        state.refreshing = initialState.refreshing;
      }, 'clearArticleScheduleContent');
    },
  },

  reset: () => resetStore(initialState, set),
});

const useArticleScheduleContentStore = createStore<IArticleScheduleContentState>(articleScheduleContentStore);

export default useArticleScheduleContentStore;
