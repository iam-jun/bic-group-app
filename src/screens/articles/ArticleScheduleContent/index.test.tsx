import React from 'react';

import { renderWithRedux } from '~/test/testUtils';
import ArticleScheduleContent from './index';

describe('ArticleScheduleContent screen', () => {
  const props = { route: { params: { groupId: '123' } } };

  it('render correctly', () => {
    const rendered = renderWithRedux(
      <ArticleScheduleContent {...props} />,
    );

    const containerView = rendered.queryByTestId('article_schedule_content');
    expect(containerView).not.toBeNull();
  });
});
