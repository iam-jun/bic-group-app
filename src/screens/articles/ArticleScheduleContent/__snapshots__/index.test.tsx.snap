// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArticleScheduleContent screen render correctly 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#FFFFFF",
      },
      Object {
        "flex": 1,
      },
      Object {
        "backgroundColor": "#EAEDF2",
        "flex": 1,
        "paddingBottom": 0,
      },
    ]
  }
  testID="article_schedule_content"
>
  <View
    style={
      Object {
        "zIndex": 2,
      }
    }
    testID="header"
  >
    <View
      animatedStyle={
        Object {
          "value": Object {},
        }
      }
      collapsable={false}
      style={
        Object {
          "alignItems": "flex-end",
          "backgroundColor": "#FFFFFF",
          "elevation": 2,
          "flexDirection": "row",
          "minHeight": 44,
          "paddingTop": 0,
          "paddingVertical": 4,
          "shadowColor": "rgba(0, 0, 0, 1)",
          "shadowOffset": Object {
            "height": 2,
            "width": 0,
          },
          "shadowOpacity": 0.12,
          "shadowRadius": 1.5,
        }
      }
      testID="header.content"
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#FFFFFF",
            "flex": 1,
            "flexDirection": "row",
            "overflow": "hidden",
            "paddingLeft": 8,
            "paddingRight": 8,
          }
        }
      >
        <View
          accessibilityState={
            Object {
              "disabled": false,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          hitSlop={
            Object {
              "bottom": 20,
              "left": 20,
              "right": 20,
              "top": 20,
            }
          }
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
          testID="header.back.button"
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "flexDirection": "row",
                },
                Object {
                  "alignItems": "center",
                  "height": 44,
                  "justifyContent": "center",
                  "padding": 12,
                  "width": 44,
                },
                Object {
                  "backgroundColor": undefined,
                },
              ]
            }
          >
            <View
              style={
                Array [
                  undefined,
                  undefined,
                  undefined,
                ]
              }
            >
              <View
                testID="header.back"
              />
            </View>
          </View>
        </View>
        <View
          animatedStyle={
            Object {
              "value": Object {},
            }
          }
          collapsable={false}
          style={
            Object {
              "flex": 1,
              "justifyContent": "center",
              "paddingVertical": 8,
            }
          }
        >
          <View
            accessibilityState={
              Object {
                "disabled": true,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={false}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "opacity": 1,
              }
            }
          >
            <Text
              numberOfLines={1}
              style={
                Array [
                  Object {
                    "color": "#181C32",
                    "fontFamily": "BeVietnamPro-SemiBold",
                    "fontSize": 15,
                  },
                  Object {},
                  Object {
                    "marginRight": 8,
                  },
                ]
              }
              testID="header.text"
            >
              Schedule Content
            </Text>
          </View>
        </View>
      </View>
      <View
        style={
          Object {
            "backgroundColor": "#FFFFFF",
            "height": 0,
            "position": "absolute",
            "top": 0,
            "width": "100%",
          }
        }
      />
    </View>
  </View>
  <RCTScrollView
    ItemSeparatorComponent={[Function]}
    ListEmptyComponent={[Function]}
    ListFooterComponent={[Function]}
    ListHeaderComponent={[Function]}
    data={Array []}
    getItem={[Function]}
    getItemCount={[Function]}
    keyExtractor={[Function]}
    onContentSizeChange={[Function]}
    onEndReached={[Function]}
    onEndReachedThreshold={0.2}
    onLayout={[Function]}
    onMomentumScrollBegin={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    refreshControl={
      <RefreshControlMock
        onRefresh={[Function]}
        refreshing={true}
        tintColor="#A4A9B7"
      />
    }
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={Array []}
    style={
      Object {
        "flex": 1,
      }
    }
    testID="article_schedule_content.list"
    viewabilityConfigCallbackPairs={Array []}
  >
    <RCTRefreshControl />
    <View>
      <View
        onLayout={[Function]}
      >
        <View
          style={
            Object {
              "height": 16,
              "paddingHorizontal": undefined,
              "paddingVertical": undefined,
              "width": undefined,
            }
          }
        />
      </View>
      <View
        onLayout={[Function]}
      >
        <View />
      </View>
    </View>
  </RCTScrollView>
</View>
`;
